include/sampling/sampling.hpp

```diff
@@ -434,169 +373,162 @@ crhmc_sampling <
 >(randPoints, P, rng, walkL, numpoints, nburns, *F, *f, zerof, simdLen, raw_output);
 }
 }
-template
-<
-        typename WalkTypePolicy,
-        typename PointList,
-        typename Polytope,
-        typename RandomNumberGenerator,
-        typename NT,
-        typename Point
->
-void exponential_sampling(PointList &randPoints,
-                          Polytope &P,
-                          RandomNumberGenerator &rng,
-                          const unsigned int &walk_len,
-                          const unsigned int &rnum,
-                          const Point &c,
-                          const NT &a,
-                          const Point &starting_point,
-                          unsigned int const& nburns)
-{
 
-    typedef typename WalkTypePolicy::template Walk
-            <
-                    Polytope,
-                    RandomNumberGenerator
-            > walk;
 
-    PushBackWalkPolicy push_back_policy;
-
-    Point p = starting_point;
-
-    typedef ExponentialRandomPointGenerator <walk> RandomPointGenerator;
-    if (nburns > 0) {
-        RandomPointGenerator::apply(P, p, c, a, nburns, walk_len, randPoints,
-                                    push_back_policy, rng);
-        randPoints.clear();
-    }
-    RandomPointGenerator::apply(P, p, c, a, rnum, walk_len, randPoints,
-                                push_back_policy, rng);
-}
 
 
 template <
-        typename PointList,
-        typename Polytope,
-        typename RandomNumberGenerator,
-        typename WalkTypePolicy,
-        typename NT,
-        typename Point
-        >
-void exponential_sampling(PointList &randPoints,
-                       Polytope &P,
-                       RandomNumberGenerator &rng,
-                       WalkTypePolicy &WalkType,
-                       const unsigned int &walk_len,
-                       const unsigned int &rnum,
-                       const Point &c,
-                       const NT &a,
-                       const Point &starting_point,
-                       unsigned int const& nburns)
-{
-
-    typedef typename WalkTypePolicy::template Walk
-            <
-                    Polytope,
-                    RandomNumberGenerator
-            > walk;
+    typename WalkTypePolicy,
+    typename PointList,
+    typename Polytope,
+    typename RandomNumberGenerator,
+    typename NT,
+    typename Point
+>
+void exponential_sampling(
+    PointList &randPoints,
+    Polytope &P,
+    RandomNumberGenerator &rng,
+    const unsigned int &walk_len,
+    const unsigned int &rnum,
+    const Point &c,
+    const NT &a,
+    Point &starting_point, 
+    unsigned int const& nburns
+) {
```

Comment:
looks like a new style of code, I suggest using the style of most of the functions in volesti